src/confcom/azext_confcom/fragment_util.py

```diff
@@ -25,16 +30,33 @@ def combine_fragments_with_policy(all_fragments):
     return out_fragments
 
 
-def get_all_fragment_contents(fragment_imports):
+def get_all_fragment_contents(
+    image_names: List[str] = [],
+    fragment_imports: List[dict] = []
+) -> List[str]:
+    # was getting errors with pass by reference so we need to copy it
+    copied_fragment_imports = copy.deepcopy(fragment_imports)
+
     fragment_feeds = [
         case_insensitive_dict_get(fragment, config.POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS_FEED)
-        for fragment in fragment_imports
+        for fragment in copied_fragment_imports
     ]
 
     all_fragments_contents = []
-    cose_proxy = CoseSignToolProxy()
+    # get all the image attached fragments
+    for image in image_names:
+        # TODO: make sure this doesn't error out if the images aren't in a registry. This will probably be in the discover function
+        fragments, feeds = oras_proxy.pull_all_image_attached_fragments(image)
+        for fragment, feed in zip(fragments, feeds):
+            if feed in fragment_feeds:
+                all_fragments_contents.append(fragment)
+            else:
+                logger.warning("Fragment feed %s not in list of feeds to use. Skipping fragment.", feed)
 
-    for fragment in fragment_imports:
+    cose_proxy = CoseSignToolProxy()
+    # get all the local fragments
+    for fragment in copied_fragment_imports:
+        contents = []
         # pull locally if there is a path, otherwise pull from the remote registry
         if (
             config.POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS_PATH in fragment and
```

Comment:
this might be a little shorter: `if fragment.get(config.POLICY_FIELD_CONTAINERS_ELEMENTS_REGO_FRAGMENTS_PATH):`